TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -48,25 +44,52 @@ private static void checkReadOnly(Node node, [NotNull] Node receiving)
 
     class DataDefinitionChecker
     {
-        public static void OnNode(Node node, DataDescriptionEntry dataEntry = null)
+        public static void OnNode(DataDefinition dataDefinition, DataDescriptionEntry dataEntry = null)
         {
-            if (dataEntry == null && node is DataDefinition)
+            if (dataEntry == null)
             {
-                dataEntry = node.CodeElement as DataDescriptionEntry;
+                dataEntry = dataDefinition.CodeElement as DataDescriptionEntry;
```

Comment:
@smedilol Does the check '"Variable with usage COMP-1 and COMP-2 cannot have a PICTURE"' is TC-specific ? If not, we should move the whole method into `Visit(DataDescription)` in `CrossChecker`

Reply:
This test is also valid for Cobol.

Reply:
Code moved to `CrossChecker`